### PR TITLE
Add generalized crossplatform support for Hyperion serializer.

### PR DESCRIPTION
**Pull request by Arkatufus**

### CODE REVIEW

**Comment by Arkatufus** [comment]
Need to wait for a new Hyperion changes first.

**Comment by Aaronontheweb** [comment]
@Arkatufus those new Hyperion changes are ready now, yes?

**Comment by Arkatufus** [comment]
They are ready, but they're not merged yet. These changes would need the new Hyperion nuget package.

**Comment by Arkatufus** [comment]
Ok, this is ready for review.

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
LGTM, but need to add some user-facing documentation to here: https://github.com/akkadotnet/akka.net/blob/dev/docs/articles/networking/serialization.md#how-to-setup-hyperion-as-default-serializer

File: src/contrib/serializers/Akka.Serialization.Hyperion/HyperionSerializerSetup.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
+﻿using System;
```
Comment (Aaronontheweb) [inline src/contrib/serializers/Akka.Serialization.Hyperion/HyperionSerializerSetup.cs:1]: Need to add copyright header

File: src/contrib/serializers/Akka.Serialization.Hyperion/HyperionSerializerSetup.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            VersionTolerance = versionTolerance;
+            KnownTypesProvider = knownTypesProvider;
+            PackageNameOverrides = packageNameOverrides;
+        }
+
+        /// <summary>
+        /// When true, it tells <see cref="HyperionSerializer"/> to keep
+        /// track of references in serialized/deserialized object graph.
+        /// </summary>
+        public readonly Option<bool> PreserveObjectReferences;
```
Comment (Aaronontheweb) [inline src/contrib/serializers/Akka.Serialization.Hyperion/HyperionSerializerSetup.cs:38]: Make these properties rather than fields

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**